<launch>
	
	<!-- RTAB-MAP LOOP CLOSURE DETECTION VERSION -->
	<!-- WARNING : Database is automatically deleted on each startup -->
	<!--           See "delete_db_on_start" option below... -->
		
	<!-- args: "delete_db_on_start" and "udebug" -->
	<node name="rtabmap" pkg="rtabmap" type="rtabmap" output="screen" args="--delete_db_on_start">
	
		<param name="config_path" value="~/.rtabmap/demo.ini" type="string"/>
		<param name="working_directory" value="~/.rtabmap" type="string"/>  
	</node>
	
	<node name="rtabmapviz" pkg="rtabmap" type="rtabmapviz" output="screen"
	 args="-d ~/.rtabmap/demo.ini"/>
	
	<!-- uimage pkg is from the UtiLite stack : svn checkout http://utilite.googlecode.com/svn/trunk/ros-pkg utilite -->
	<!-- When parameter video_or_images_path is set, the camera uses the directory of images or the video file -->
	<node name="camera" pkg="uimage" type="camera" output="screen">
		<remap from="/camera/image" to="/image"/>
		<param name="device_id" value="0" type="int"/>
		<param name="video_or_images_path" value="$(find rtabmap_lib)/build/rtabmap-svn/bin/data/samples" type="string"/>
		<param name="frame_rate" value="2.0" type="double"/>
		<param name="width" value="0" type="int"/>
		<param name="height" value="0" type="int"/>
		<param name="auto_restart" value="false" type="bool"/> <!-- Process only one time the data set -->
	</node>
</launch>
